src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -5529,9 +5627,33 @@ instruct compareAndExchangeB(iRegINoSp res, indirect mem, iRegI_R12 oldval, iReg
   ins_pipe(pipe_slow);
 %}
 
-instruct compareAndExchangeS(iRegINoSp res, indirect mem, iRegI_R12 oldval, iRegI_R13 newval,
+instruct compareAndExchangeB(iRegINoSp res, indirect mem, iRegI_R12 oldval, iRegI_R13 newval, rFlagsReg cr)
+%{
+  predicate(UseZabha && UseZacas);
+
+  match(Set res (CompareAndExchangeB mem (Binary oldval newval)));
+
+  ins_cost(LOAD_COST + STORE_COST + BRANCH_COST * 3 + ALU_COST * 5);
+
+  effect(TEMP_DEF res, KILL cr, USE_KILL oldval, USE_KILL newval);
```

Comment:
It's similar to word and doubleword CompareAndExchange cases when we have UseZabha and UseZacas. We have a more simpler `effect(TEMP_DEF res)` for word and doubleword CompareAndExchange cases. So I think we can simplify this effect as well. Similar for `compareAndExchangeS`, `compareAndExchangeBAcq` and `compareAndExchangeSAcq`.

And after having another look at the code of `MacroAssembler::cmpxchg`, seems we don't even need this `effect(TEMP_DEF res)` for all these cases including the existing word and doubleword CompareAndExchange cases.